opal/corelib/basic_object.rb

```diff
@@ -34,6 +34,10 @@ def !
     false
   end
 
+  def !=(other)
+    self != other
```

Comment:
Yeah, that would fail in the general case. For subclasses that redefine `==`, we want `!=` to behave correctly.